redash/query_runner/activedata.py

```diff
@@ -0,0 +1,180 @@
+
+    @classmethod
+    def type(cls):
+        return "activedata"
+
+    @classmethod
+    def enabled(cls):
+        return True
+
+    def _get_tables(self, schema):
+        query = {
+            "from": "meta.columns",
+            "select": [
+                "name",
+                "table"
+            ],
+            "where": {"not": {"prefix": {"es_index": "meta."}}},
+            "limit": 1000,
+            "format": "list"
+        }
+        results = self.run_jx_query(query, None)
+
+        for row in results['data']:
+            table_name = row['table']
+
+            if table_name not in schema:
+                schema[table_name] = {'name': table_name, 'columns': []}
+
+            schema[table_name]['columns'].append(row['name'])
+
+        return [{'name': r['name'], 'columns': sorted(r['columns'])} for r in schema.values()]
+
+    def run_jx_query(self, query, user):
+        data = json.dumps(query, ensure_ascii=False)
+        result = requests.post(self.configuration['host_url']+"/query", data=data)
+        response = json.loads(result.content)
+
+        if response.get('type') == "ERROR":
+            cause = find_cause(response)
+            raise Exception(cause)
+        return response
+
+    def run_query(self, annotated_query, user):
+        request = {}
+        comment, request["sql"] = annotated_query.split("*/", 2)
+        meta = request['meta'] ={}
+        for kv in comment.strip()[2:].split(","):
+            k, v = map(unicode.strip, kv.split(":"))
+            meta[k] = v
+
+        logger.debug("Send ActiveData a SQL query: %s", request['sql'])
+        data = json.dumps(request, ensure_ascii=False)
+        result = requests.post(self.configuration['host_url']+"/sql", data=data)
+        response = json.loads(result.content)
+
+        if response.get('type') == "ERROR":
+            cause = find_cause(response)
+            return None, cause
+
+        output = normalize(response)
+        json_data = json.dumps(output, cls=JSONEncoder)
+        return json_data, None
+
+
+
+def normalize(table):
+    columns = {}  # MAP FROM name TO (MAP FROM type TO (full_name))
+    output = []
+
+    def get_unique_name(name, type):
+        all_types = columns.get(name)
+        if all_types is None:
+            all_types = columns[name] = {}
+        specific_type = all_types.get(type)
+        if specific_type is None:
+            if all_types:
+                specific_type = all_types[type] = name + "." + type
+            else:
+                specific_type = all_types[type] = name
+        return specific_type
+
+    for r in table['data']:
+        new_row = {}
+        for i, cname in enumerate(table['header']):
+            val = r[i]
+            if val == None:
+                continue
+            type_ = val.__class__
+            if type_ in [dict, list]:
+                val = json.dumps(val, cls=JSONEncoder)
+            col = get_unique_name(cname, types_map[type_])
```

Comment:
If you remove `type_` above then just using `types_map.get(type(val), TYPE_STRING)` would work here. I think using `type` and `isinstance` is preferred.